File: 383.go
package leetcode

func canConstruct(ransomNote string, magazine string) bool {
	m := len(ransomNote)
	n := len(magazine)
	if m > n {
		return false
	}
	cnt := make([]int, 26)
	for i := 0; i < m; i++ {
		cnt[ransomNote[i]-97]--
	}
	for i := 0; i < n; i++ {
		cnt[magazine[i]-97]++
	}
	for _, v := range cnt {
		if v < 0 {
			return false
		}
	}
	return true
}

File: 383.py
import collections


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        # m, n = len(ransomNote), len(magazine)
        # if m > n:
        #     return False
        # cnt = [0] * 26
        # for a, b in zip_longest(ransomNote, magazine):
        #     if a is not None:
        #         cnt[ord(a)-ord('a')] -= 1
        #     cnt[ord(b)-ord('a')] += 1
        # return all(x >= 0 for x in cnt)

        if len(ransomNote) > len(magazine):
            return False
        return not collections.Counter(ransomNote) - collections.Counter(magazine)
